### PR TITLE
Implement Point#mul

### LINKED ISSUE #306: OpenSSL::PKey::EC::Point missing `mul` method
It looks like the `OpenSSL::PKey::EC::Point` [class](https://github.com/jruby/jruby-openssl/blob/master/src/main/java/org/jruby/ext/openssl/PKeyEC.java#L895) is missing the `mul` method.  It does look like bouncy castle's ECPoint class does support [multiply](https://javadoc.io/doc/org.bouncycastle/bcprov-jdk15on/latest/org/bouncycastle/math/ec/ECPoint.html#multiply-java.math.BigInteger-). 

## Steps to reproduce
```ruby
require 'openssl'
group = OpenSSL::PKey::EC::Group.new('prime256v1')
new_point = group.generator.mul(2)
```

## Expected Result
```
# <OpenSSL::PKey::EC::Point>
```

In cRuby:

```ruby
require 'openssl'
group = OpenSSL::PKey::EC::Group.new('prime256v1')
new_point = group.generator.mul(2) # returns <OpenSSL::PKey::EC::Point:0x000000010dc80218 @group=#<OpenSSL::PKey::EC::Group:0x000000010dc80268>>
```

## Actual result
```
 undefined method `mul' for #<OpenSSL::PKey::EC::Point:0x1f39b3ba> (NoMethodError)
```

## Version
* JRuby: 9.4.5.0
* jruby-openssl: jruby-openssl-0.14.6-java

### PR DESCRIPTION
**Pull request by headius**
Work in progress. Fixes #306.

### CODE REVIEW

File: src/main/java/org/jruby/ext/openssl/PKeyEC.java [hunk 1]
```diff
@@ -1059,6 +1071,61 @@ public IRubyObject inspect() {
...
+                    return initialize(context, args[0]);
+                case 2:
+                    return initialize(context, args[0], args[1]);
+                default:
+                    throw context.runtime.newArgumentError(args.length, 1);
+            }
+        }
+
+        @JRubyMethod(name = "mul")
+        public IRubyObject mul(final ThreadContext context, final IRubyObject[] args) {
```
Comment (alextwoods) [inline src/main/java/org/jruby/ext/openssl/PKeyEC.java:1089]: I think there is at least one other math operation on `ECPoint` that this change now makes it easy to support: [add](https://docs.ruby-lang.org/en/3.3/OpenSSL/PKey/EC/Point.html#method-i-add)

Comment (headius) [reply src/main/java/org/jruby/ext/openssl/PKeyEC.java:1077]: Yes I'll do add in another PR!

**Comment by headius** [comment]
Initial implementation is in place and appears to work for your simple example:

```
[] jruby-openssl $ jruby -Ilib blah.rb
"\x04|\xF2{\x18\x8D\x03O~\x8AR8\x03\x04\xB5\x1A\xC3\xC0\x89i\xE2w\xF2\e5\xA6\vH\xFCGf\x99x\awU\x10\xDB\x8E\xD0@)=\x9A\xC6\x9Ft0\xDB\xBA}\xAD\xE6<\xE9\x82)\x9E\x04\xB7\x9D\"xs\xD1"
[] jruby-openssl $ ruby33 blah.rb  
"\x04|\xF2{\x18\x8D\x03O~\x8AR8\x03\x04\xB5\x1A\xC3\xC0\x89i\xE2w\xF2\e5\xA6\vH\xFCGf\x99x\awU\x10\xDB\x8E\xD0@)=\x9A\xC6\x9Ft0\xDB\xBA}\xAD\xE6<\xE9\x82)\x9E\x04\xB7\x9D\"xs\xD1"
```

I'm comparing uncompressed octet string output from each here. Better confirmation would be to run the related CRuby openssl tests, but there are other issues preventing us from running `test_ec_point_mul` and `test_ec_point_add` (4-arg constructor for Group is not implemented). This is at least a good start!

**Comment by headius** [comment]
@alextwoods This now has `add`. Would you be able to test this branch against your code, or at least a more complex example? It will take more work to fill out `Group` logic to support the test in CRuby.

@kares Review please!

**Comment by alextwoods** [comment]
This works great!  The AWS SDK For Ruby has a fairly extensive test suite for a signing algorithm based off of ECC and relies on the `mul` method (sigv4a) - this change passes all of our tests :-)

**Comment by headius** [comment]
> this change passes all of our tests

Gotta admit I'm surprised! But I think we can merge this in if @kares is ok with it. Fair warning: it's probably not going to be live until 0.15.0 releases unless we backport to 0.14 and do another release there.

File: src/main/java/org/jruby/ext/openssl/PKeyEC.java [hunk 2]
```diff
@@ -931,44 +935,54 @@ private static RaiseException newError(final Ruby runtime, final String message)
...
+
+            if ( groupOrPoint instanceof Point) {
+                this.group = ((Point) groupOrPoint).group;
+                this.point = ((Point) groupOrPoint).point;
+                return true;
+            }
+
+            if ( groupOrPoint instanceof Group) {
+                this.group = (Group) groupOrPoint;
+                this.point = (ECPoint) ((Group) groupOrPoint).generator(context);
```
Comment (kares) [inline src/main/java/org/jruby/ext/openssl/PKeyEC.java:979]: as hinted by the failing tests, this does not return an `ECPoint` but rather the native (Ruby) `org.jruby.ext.openssl.PKeyEC$Point` instance

Comment (headius) [reply src/main/java/org/jruby/ext/openssl/PKeyEC.java]: Fixed by getting paramSpec from the group and using that to get the generator point. Seems to be green now.

**Comment by headius** [comment]
I would like to add `Point#add` as well but hoping to get the related tests running first. That depends on getting #309 working.

**Pull request merged**